src/main/java/core/basesyntax/App.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import core.basesyntax.data.FileService;
+import core.basesyntax.data.FileServiceImpl;
+import core.basesyntax.data.csv.FileReader;
+import core.basesyntax.data.csv.FileWriter;
+import core.basesyntax.data.csv.impl.CsvFileReaderImpl;
+import core.basesyntax.data.csv.impl.CsvFileWriterImpl;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ReportGeneratorImpl;
+
+public class App {
+    private static final String INPUT_FILE_PATH = "src/main/resources/input.csv";
+    private static final String OUTPUT_FILE_PATH = "src/main/resources/output.csv";
+    private static final String OUTPUT_FILE_HEADER_ROW = "fruit,quantity";
+
+    public static void main(String[] args) {
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
+
+        FileWriter csvWriter = new CsvFileWriterImpl(OUTPUT_FILE_PATH, OUTPUT_FILE_HEADER_ROW);
+        FileReader csvReader = new CsvFileReaderImpl(INPUT_FILE_PATH);
+        FileService fileService = new FileServiceImpl(csvWriter, csvReader);
+
+        var transactionHistory = fileService.getTransactions();
+        var report = reportGenerator.generateReport(transactionHistory);
```

Comment:
I wouldn't use var where it's not obvious what is returned, use it to reduce verbosity in cases like Product product = new Product()
The best practice concerning the use of var is to use it when the right-hand side of the statement makes it clear what the type is.